migrations/20210612_initial.sql

```diff
@@ -194,85 +194,107 @@ CREATE TABLE connections ( -- all SMP agent connections
     DEFERRABLE INITIALLY DEFERRED
 );
 
-CREATE TABLE events ( -- messages received by the agent, append only
-  event_id INTEGER PRIMARY KEY,
-  agent_msg_id INTEGER NOT NULL, -- internal message ID
+CREATE TABLE messages ( -- messages received by the agent, append only
+  message_id INTEGER PRIMARY KEY,
+  agent_msg_id INTEGER NOT NULL, -- internal agent message ID
   external_msg_id INTEGER NOT NULL, -- external message ID (sent or received)
-  agent_meta TEXT NOT NULL, -- JSON with timestamps etc. sent in MSG
+  agent_meta TEXT, -- JSON with timestamps etc. sent in MSG, NULL for sent
   connection_id INTEGER NOT NULL REFERENCES connections,
-  received INTEGER NOT NULL, -- 0 for received, 1 for sent
-  chat_event_id INTEGER,
-  continuation_of INTEGER, -- references chat_event_id, but can be incorrect
-  event_type TEXT NOT NULL, -- event type - see protocol/types.ts
-  event_encoding INTEGER NOT NULL, -- format of event_body: 0 - binary, 1 - text utf8, 2 - JSON (utf8)
-  content_type TEXT NOT NULL, -- content type - see protocol/types.ts
-  event_body BLOB, -- agent message body as sent
-  event_hash BLOB NOT NULL,
-  integrity TEXT NOT NULL DEFAULT '',
+  msg_sent INTEGER NOT NULL, -- 0 for received, 1 for sent
+  chat_msg_id INTEGER NOT NULL, -- sent as part of the message
+  chat_msg_event TEXT NOT NULL, -- message event type (the constructor of ChatMsgEvent)
+  msg_body BLOB, -- agent message body as sent
+  msg_body_hash BLOB NOT NULL,
+  msg_integrity TEXT NOT NULL DEFAULT '',
   created_at TEXT NOT NULL DEFAULT (datetime('now'))
 );
 
-CREATE INDEX events_external_msg_id_index ON events (connection_id, external_msg_id);
+CREATE INDEX messages_agent_msg_id_index ON messages (connection_id, agent_msg_id);
 
-CREATE TABLE event_body_parts (
-  event_body_part_id INTEGER PRIMARY KEY,
-  event_id REFERENCES events,
-  full_size INTEGER NOT NULL,
-  part_status TEXT, -- full, partial
-  content_type TEXT NOT NULL,
-  event_part BLOB
-);
+CREATE INDEX messages_external_msg_id_index ON messages (connection_id, external_msg_id);
 
-CREATE TABLE contact_profile_events (
-  event_id INTEGER NOT NULL UNIQUE REFERENCES events,
+CREATE TABLE contact_profile_messages (
+  message_id INTEGER NOT NULL UNIQUE REFERENCES messages,
   contact_profile_id INTEGER NOT NULL REFERENCES contact_profiles
 );
 
-CREATE TABLE group_profile_events (
-  event_id INTEGER NOT NULL UNIQUE REFERENCES events,
+CREATE TABLE group_profile_messages (
+  message_id INTEGER NOT NULL UNIQUE REFERENCES messages,
   group_profile_id INTEGER NOT NULL REFERENCES group_profiles
 );
 
-CREATE TABLE group_events (
-  event_id INTEGER NOT NULL UNIQUE REFERENCES events,
+CREATE TABLE direct_messages (
+  message_id INTEGER NOT NULL UNIQUE REFERENCES messages,
+  contact_id INTEGER NOT NULL REFERENCES contacts ON DELETE RESTRICT,
+  msg_sent INTEGER -- 1 for sent, 0 for received
```

Comment:
ambiguous since we also have SENT notification - the field is named as if it's a flag relating to that - rename to `snd_msg`?